File: yunj/level1/8_Workout_clothes.java
import java.util.ArrayList;
import java.util.Collections;


class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
       ArrayList<Integer> lostArr = new ArrayList<>();
        ArrayList<Integer> reserveArr = new ArrayList<>();
        ArrayList<Integer> safe = new ArrayList<>();

        for(int l : lost){
            lostArr.add(l);
        }
        for(int r : reserve){
            reserveArr.add(r);
        }

        for (int l : lostArr) {
            if(reserveArr.contains(l)){
                safe.add(l);
                reserveArr.remove(new Integer(l));
            }
        }

        for (int l : lostArr) {
            if (safe.contains(l)){
                continue;
            }
            if(reserveArr.contains(l)){
                safe.add(l);
                reserveArr.remove(new Integer(l));
                continue;
            }
            if (reserveArr.contains(l+1) ^ reserveArr.contains(l-1)){
                if (reserveArr.contains(l+1)) {
                    safe.add(l);
                    reserveArr.remove(new Integer(l+1));
                } else {
                    safe.add(l);
                    reserveArr.remove(new Integer(l-1));
                }
                continue;
            }
            if(reserveArr.contains(l+1) && reserveArr.contains(l-1)){
                safe.add(l);
                reserveArr.remove(new Integer(l-1));
            }
        }


        int answer = n - (lostArr.size() - safe.size());
        return answer;
    }
}

File: sangjin/level1/8_workout_clothes.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = n - lost.size();
    vector<int> need(n+2);

    for(int i = 1; i < lost.size(); i++){
        int index = lost[i];

        need[index] == 1;
    }

    for(int i = 0; i < reserve.size(); i++){
        int index = reserve[i];
        if(need[index] != 0){
            need[index] = 0;
            answer++;
            reserve.erase(reserve.begin() + i);
            i--;
        }
    }

    for(int i = 0; i < reserve.size(); i++){
        int index = reserve[i];
        int seq[] = {-1, 1};

        for(int j = 0; j < 3; j++){
            if(need[index + seq[j]] != 0){
                need[index + seq[j]] = 0;
                answer++;
                break;
            }
        }
    }

    return answer;
}
